src/states/elder/mod.rs

```diff
@@ -1323,6 +1329,33 @@ impl Elder {
 
         SignedRoutingMessage::new(msg, secret_key, pk_set, proof)
     }
+
+    // Checks whether the message is trusted.
+    fn check_signed_message_trust(
+        &self,
+        msg: &SignedRoutingMessage,
+    ) -> Result<TrustStatus, RoutingError> {
+        match msg.check_trust(&self.chain) {
+            Ok(()) => Ok(TrustStatus::Known),
+            Err(ProofError::TooNew) if msg.routing_message().dst.is_multiple() => {
+                // Proof is too new which can only happen if we've been already demoted but are
+                // lagging behind (or the sender is faulty/malicious). We can't handle the message
+                // ourselves but the other elders likely can.
+                Ok(TrustStatus::Unknown)
```

Comment:
Why not return the same error `Err(RoutingError::UntrustedMessage)`?
The only difference we care about is probably the log or panic?